library/std/src/sys/uefi/time.rs

```diff
@@ -0,0 +1,227 @@
+use super::common;
+use crate::mem::MaybeUninit;
+use crate::ptr::NonNull;
+use crate::sys_common::mul_div_u64;
+use crate::time::Duration;
+
+use r_efi::protocols::timestamp;
+
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+pub struct Instant(Duration);
+
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+pub struct SystemTime(Duration);
+
+pub const UNIX_EPOCH: SystemTime = SystemTime(Duration::ZERO);
+
+const NS_PER_SEC: u64 = 1_000_000_000;
+const SEC_IN_MIN: u64 = 60;
+const SEC_IN_HOUR: u64 = SEC_IN_MIN * 60;
+const SEC_IN_DAY: u64 = SEC_IN_HOUR * 24;
+const SEC_IN_YEAR: u64 = SEC_IN_DAY * 365;
+
+impl Instant {
+    pub fn now() -> Instant {
+        if let Ok(handles) = common::locate_handles(timestamp::PROTOCOL_GUID) {
+            // First try using `EFI_TIMESTAMP_PROTOCOL` if present
```

Comment:
A protocol can technically be installed or uninstalled between calls to this function. Thus if the `EFI_TIMESTAMP_PROTOCOL` is not present the first time, it does not imply that it will be absent in the subsequent calls (and vice-versa).